src/main/java/net/dries007/tfc/common/blockentities/BloomeryBlockEntity.java

```diff
@@ -1,25 +1,233 @@
+                        {
+                            bloomery.dumpItems();
+                        }
+                    }
+                    if (bloomery.cachedRecipe != null)
+                    {
+                        level.setBlockAndUpdate(bloomery.getInternalBlock(), TFCBlocks.BLOOM.get().defaultBlockState());
+                        BloomBlockEntity bloom = Helpers.getBlockEntity(level, bloomery.getInternalBlock(), BloomBlockEntity.class);
+                        if (bloom != null)
+                        {
+                            bloom.setBloom(bloomery.cachedRecipe.getResultItem(bloomery.oreStacks));
+                        }
+                    }
+
+                    bloomery.oreStacks.clear();
+                    bloomery.fuelStacks.clear();
+                    bloomery.cachedRecipe = null; // Clear recipe
+
+                    bloomery.updateSlagBlock(false);
+                    state = state.setValue(BloomeryBlock.LIT, false);
+                    level.setBlockAndUpdate(pos, state);
+                }
+            }
+
+            // Update multiblock status
+            int newMaxItems = BloomeryBlock.getChimneyLevels(level, bloomery.getInternalBlock()) * 8;
+            Direction direction = state.getValue(BloomeryBlock.FACING);
+            //todo: isFormed may need to be either static, or a BlockEntity method, as this does not resolve
+            //would it be better to do all of this in Block#randomTick, like the forge block does? - would also solve above problem
+            if (!TFCBlocks.BLOOMERY.isFormed(level, bloomery.getInternalBlock(), direction))
+            {
+                newMaxItems = 0;
+            }
+
+            bloomery.maxFuel = newMaxItems;
+            bloomery.maxOre = newMaxItems;
+            boolean turnOff = false;
+            while (bloomery.maxOre < bloomery.oreStacks.size())
+            {
+                turnOff = true;
+                // Structure lost one or more chimney levels
+                Helpers.spawnItem(level, bloomery.getExternalBlock(), bloomery.oreStacks.get(0));
+                bloomery.oreStacks.remove(0);
+                bloomery.markForSync();
+            }
+            while (bloomery.maxFuel < bloomery.fuelStacks.size())
+            {
+                turnOff = true;
+                Helpers.spawnItem(level, bloomery.getExternalBlock(), bloomery.fuelStacks.get(0));
+                bloomery.fuelStacks.remove(0);
+                bloomery.markForSync();
+            }
+            // Structure became compromised, unlit if needed
+            if (turnOff && state.getValue(BloomeryBlock.LIT))
+            {
+                state = state.setValue(BloomeryBlock.LIT, false);
+                level.setBlockAndUpdate(pos, state);
+            }
+            //todo: also will not resolve - static? move all this to Block#randomTick?
+            if (!TFCBlocks.BLOOMERY.canGateStayInPlace(level, pos, direction.getAxis()))
```

Comment:
I'm not exactly sure how this even compiles?
Let's do a static method and call it like BloomeryBlock#canGateStayInPlace